### PR TITLE
Allow to run official ASPECT tester at home

### PR DESCRIPTION
**Pull request by gassmoeller**
This is related to #2143 and allows everybody to run the official ASPECT tester setup (currently only `aspect-8.5-gcc-hd`) on their local machine. This is super helpful if you know the tester is busy, or to check and update test results yourself if you are offline. It also allows with a little hack to `generate_reference_output` to only run certain tests you are interested in. This is a step towards having all the tester information as part of the official repository.

### CODE REVIEW

**Comment by gassmoeller** [comment]
Wow, it seems the tester took this PR as a personal insult :smile:, @tjhei could you check what happened?

File: cmake/compile_and_update_tests.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# This script can modify the test results in the source
+# directory to the official tester results, if docker is installed
+# on the system. Start this script by running the following command
+# after replacing ASPECT_SOURCE_DIR with the directory of your ASPECT
+# folder:
+#
+# docker run -v ASPECT_SOURCE_DIR:/home/dealii/aspect --name=aspect-tester --rm -it dealii/dealii:v8.5.0-gcc-mpi-fulldepscandi-debugrelease bash /home/dealii/aspect/cmake/compile_and_update_tests.sh
```
Comment (tjhei) [inline cmake/compile_and_update_tests.sh]: would it make sense to ask the user to execute this command in the source folder and then just specify ``$PWD`` here?

File: cmake/compile_and_update_tests.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# ASPECT folder into the container you are actually changing reference test
+# results on the host system (i.e. your computer).
+
+echo "Starting official tester and compiling source ..."
+
+SRC_PATH=`dirname $0`
+SRC_PATH=`cd $SRC_PATH/..;pwd`
+
+mkdir -p ${SRC_PATH}/tester-build
+cd ${SRC_PATH}/tester-build
```
Comment (tjhei) [inline cmake/compile_and_update_tests.sh]: you are putting the build output and temporary files in the mounted folder, meaning they will be left around after running the container, right? This mean rerunning the compilation in the container can be fast, but you are not getting a clean build...

File: cmake/compile_and_update_tests.sh (hunk 1 above)
Comment (gassmoeller) [reply cmake/compile_and_update_tests.sh]: I do not have a preference. Both is possible. In the current form it requires some user action, but it allows running it from everywhere. Do you have a preference?

File: cmake/compile_and_update_tests.sh (hunk 2 above)
Comment (gassmoeller) [reply cmake/compile_and_update_tests.sh]: True, not sure what is better. Personally, I often rerun the tester multiple times fixing a bug or a test at a time and a clean compilation every time would take forever. So the current state is how I would use it, but I am not sure if this is something everybody would do. Alternatively, we could introduce an environment variable like `ASPECT_TESTER_CLEAN_BUILD` that can be set when running the docker container to force a clean build.

**Comment by tjhei** [comment]
> "Wow, it seems the tester took this PR as a personal insult"

I triggered a retest and she reluctantly tested it now. ;-)

File: cmake/compile_and_update_tests.sh (hunk 1 above)
Comment (tjhei) [reply cmake/compile_and_update_tests.sh]: I prefer something copy&paste-able, but I don't feel strongly about it.

File: cmake/compile_and_update_tests.sh (hunk 2 above)
Comment (tjhei) [reply cmake/compile_and_update_tests.sh]: If this works reliably for you, then we should leave it as is.

**Comment by gassmoeller** [comment]
I made the change to require running it from the source directory and extended the documentation a bit. Are you ok with the changes?

**Comment by tjhei** [comment]
Looks good!

**Comment by gassmoeller** [comment]
Thanks!

**Pull request merged**